reconcile/utils/vcs.py

```diff
@@ -157,6 +169,44 @@ def get_commit_sha(
         # assume gitlab by default
         return self._gitlab_instance.get_commit_sha(ref=ref, repo_url=repo_url)
 
+    def get_commits_between(
+        self,
+        repo_url: str,
+        commit_from: str,
+        commit_to: str,
+        auth_code: Optional[HasSecret],
+    ) -> list[Commit]:
+        """
+        Return a list of commits between two commits.
+        Note, that the commit_to is included in the result list, whereas commit_from is not included.
+        """
+        commits: list[Commit] = []
+        if "github.com" in repo_url:
+            github = self._init_github(repo_url=repo_url, auth_code=auth_code)
+            data = github.compare(commit_from=commit_from, commit_to=commit_to)
+            for gh_commit in data:
+                commits.append(
+                    Commit(
+                        repo=repo_url,
+                        sha=gh_commit.sha,
+                        date=gh_commit.commit.committer.date,
+                    )
+                )
```

Comment:
since `commits` can either be in `gitlab` branch or `github` branch, not both, to make it clear, we can skip outer `commits`, do early return here using list comprehension directly.

```python
            return [
                Commit(
                    repo=repo_url,
                    sha=gh_commit.sha,
                    date=gh_commit.commit.committer.date,
                )
                for gh_commit in data
            ]
```

same as `gitlab` section below.

Reply:
Adjusted :+1: